src/polyswarm_api/types/artifact.py

```diff
@@ -191,11 +191,10 @@ def polyscore(self):
         return latest.polyscore
 
 
-
 class LocalArtifact(Hashable):
     """ Artifact for which we have local content """
     def __init__(self, path=None, content=None, artifact_name=None, artifact_type=ArtifactType.FILE,
-                 artifact=None, polyswarm=None, lookup=False, analyze=True):
+                 artifact=None, polyswarm=None, lookup=False, analyze=True, analyzers=None, features=None):
```

Comment:
I like the fact that this PR is an extra, but it is touching deep into layers of the code that should be orchestrated by the extra feature rather than being coupled to it. I feel like we are mixing responsibilities here and we should try to keep these things decoupled. Maybe we could have a `Analysis` object in another package that keeps a reference to the `LocalArtifiact` and keeps all the necessary properties/methods necessary to implement the functionality. This way, the `Analysis` object would manipulate the `LocalArtifact` and without the need for it to even know about it's existence (and it shouldn't, since it is an extra).

Reply:
This is why I opened that Jira ticket, I really would prefer for almost all of this to be separate and shared between AI and polyswarm-api. An analysis object sounds like a good idea, let me write something up and see if it’s what you had in mind.